cmd/prober/write.go

```diff
@@ -342,6 +357,149 @@ func rekorEntryRequest(cert *x509.Certificate, priv *ecdsa.PrivateKey) ([]byte,
 	return json.Marshal(pe)
 }
 
+// rekorV2Workflow emulates the typical signing workflow with Rekor v2,
+// which involves fetching and verifying a timestamp before adding an
+// entry to the transparency log
+func rekorV2Workflow(ctx context.Context, cert *x509.Certificate, priv *ecdsa.PrivateKey, rekorV2Services []root.Service, tsaServices []root.Service) error {
+	if err := ctx.Err(); err != nil {
+		return err
+	}
+
+	artifact := []byte(time.Now().String())
+	digest := sha256.Sum256(artifact)
+	sig, err := ecdsa.SignASN1(rand.Reader, priv, digest[:])
+	if err != nil {
+		return err
+	}
+
+	if err := fetchAndVerifyTimestamp(sig, tsaServices); err != nil {
```

Comment:
What do you think about calling this from `runProbers`? Rekor and timestamping are not connected in v2, and you might have a private deployment that doesn't stand up a Rekor instance but does stand up a timestamp authority for Fulcio.